extensions/amp-google-assistant-assistjs/0.1/amp-google-assistant-voice-bar.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This custom element displays a horizontal bar that consists of a Google Assisant voice button and a series of
+ * suggestion chips that enable 3P site users to interact with Google Assistant.
+ */
+
+import {Services} from '../../../src/services';
+import {createElementWithAttributes} from '../../../src/dom';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+export class AmpGoogleAssistantVoiceBar extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?AssistjsConfigService} */
+    this.configService_ = null;
+  }
+
+  /** @override */
+  async buildCallback() {
+    this.configService_ = await Services.assistjsConfigServiceForDoc(
+      this.element
+    );
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    // Set frame URL to an embed endpoint.
+    const frameUrl = `${this.configService_.getAssistjsServer()}/assist/voicebar?origin=${origin}&projectId=${this.configService_.getProjectId()}&dev=${this.configService_.getDevMode()}&hostUrl=${this.configService_.getHostUrl()}`;
```

Comment:
I actually agree that it would make it cleaner, since the iframe url creation is kept in a single place instead of mulitple copies across all widgets. Added a new method in config service to get the iframe url.